### PR TITLE
Prevent plugging an allocator twice

### PR DESCRIPTION
**Pull request by VibhuJawa**
## Description
This PR prevent plugging an allocator twice

### CODE REVIEW

File: nemo_curator/utils/script_utils.py [hunk 1]
```diff
@@ -472,6 +451,7 @@ def parse_gpu_dedup_args(description: str) -> argparse.ArgumentParser:
 
         # Set default device to GPU for dedup
         argumentHelper.parser.set_defaults(device="gpu")
+        argumentHelper.parser.set_defaults(set_torch_to_use_rmm=False)
```
Comment (VibhuJawa) [inline nemo_curator/utils/script_utils.py:454]: @ayushdg ,  This is what you wanted me switch off right ?

File: nemo_curator/utils/script_utils.py [hunk 2]
```diff
@@ -392,67 +408,30 @@ def parse_distributed_classifier_args(
...
+        argumentHelper.add_distributed_args()
+        argumentHelper.add_arg_input_data_dir(required=True)
+        argumentHelper.add_arg_output_data_dir(help="The path of the output files")
+        argumentHelper.add_arg_input_file_type()
+        argumentHelper.add_arg_input_file_extension()
+        argumentHelper.add_arg_output_file_type()
+        argumentHelper.add_arg_input_text_field()
+        argumentHelper.add_arg_enable_spilling()
+        argumentHelper.add_arg_set_torch_to_use_rmm()
+        argumentHelper.add_arg_batch_size(
```
Comment (VibhuJawa) [inline nemo_curator/utils/script_utils.py:421]: Decided to clean this up too because i was in the same fine, @ryantwolf , please take a look.

**Review by ryantwolf — COMMENTED** [review]
All the changes you made look good on my end. There are a couple of dedup scripts that manually set
```python
args.set_torch_to_use_rmm = False
```

Could you go through and remove that line since you change the default?

**Comment by VibhuJawa** [comment]
> Could you go through and remove that line since you change the default?

Done (Also made it false) in the function call to be safe.

File: nemo_curator/utils/script_utils.py (hunk 1 above)
Comment (ayushdg) [reply nemo_curator/utils/script_utils.py:454]: That's right!

**Review by ayushdg — APPROVED** [review]
Thanks for the changes!

**Pull request merged**